types/graphql-react/tsconfig.json

```diff
@@ -0,0 +1,25 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "jsx": "react"
```

Comment:
In order to use the default import of React in the test file, I added `esModuleInterop: true` here.
Because the type declaration file for `react` doesn't have a default export, you need this flag if you want to use `import React from "react"`. I'm not exactly sure why the authors of the `react` d.ts chose not to provide a default export, but there seems to be some oddness going on with whether or not the module actually has a default property at runtime or not. (cc @weswigham for thoughts).

Reply:
This is a great tip, and helped me fix an issue in [a different PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34277).

However, my challenge here is a bit different. When I run `npm run lint graphql-react`, I get the following in the console:

```
> definitely-typed@0.0.3 lint /Users/mike/Code/DefinitelyTyped
> dtslint types "graphql-react"
















Error: Errors in typescript@next for external dependencies:
../react/index.d.ts(30,22): error TS2307: Cannot find module 'csstype'.

    at /Users/mike/Code/DefinitelyTyped/node_modules/dtslint/bin/index.js:189:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/mike/Code/DefinitelyTyped/node_modules/dtslint/bin/index.js:5:58)
```

I'm on mac with node v11.12.0 and a fresh `node_modules`. Installing the module globally has no effect, and there is no `@types/csstype`. I would have dug into the react type definitions more, but this seems not to be a problem in CI. Of course, this leads to that painfully slow feedback loop: change, commit, push, wait, repeat.

Reply:
It seems like I'm not the only one with [this issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/24788), so I'll go ahead and report it over there.

Reply:
Well, I figured it out, and posted an issue in [dtslint](https://github.com/Microsoft/dtslint/issues/226)